### DISCUSSION TITLE
/zh-cn/blog/posts/buuctf_firmware

### DISCUSSION THREAD

**Discussion by giscus** [Reverier-Xu/discussions#150] [Announcements]
# BUUCTF Firmware WriteUp - Reverier's Blog

首先下载题目文件, 使用binwalk解压出固件内的文件
[Rx][•ᴗ•] >>> binwalk -e ./51475f91-7b90-41dd-81a3-8b82df4f29d0.bin

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
0             0x0             TP-Link firmware header, firmware version: 1.-20432.3, image version: "", product ID: 0x0, product version: 155254791, kernel load address: 0x0, kernel entry point: 0x80002000, kernel offset: 4063744, kernel length: 512, rootfs offset: 772784, rootfs length: 1048576, bootloader offset: 2883584, bootloader length: 0
69424         0x10F30         Certificate in DER format (x509 v3), header length: 4, sequence length: 64
94080         0x16F80         U-Boot version string, "U-Boot 1.1.4 (Aug 26 2013 - 09:07:51)"
94256         0x17030         CRC32 polynomial table, big endian
131584        0x20200         TP-Link firmware header, firmware version: 0.0.3, image version: "", product ID: 0x0, product version: 155254791, kernel load address: 0x0, kernel entry point: 0x80002000, kernel offset: 3932160, kernel length: 512, rootfs offset: 772784, rootfs length: 1048576, bootloader offset: 2883584, bootloader length: 0
132096        0x20400         LZMA compressed data, properties: 0x5D, dictionary size: 33554432 bytes, uncompressed size: 2203728 bytes
1180160       0x120200        Squashfs filesystem, little endian, version 4.0, compression:lzma, size: 2774624 bytes, 519 inodes, blocksize: 131072 bytes, created: 2015-04-13 09:35:04

https://blog.woooo.tech/posts/buuctf_firmware/

**Comment by mark0smith** [comment]
请问文章最后的IDA使用的哪个版本、有用什么优化主题的插件吗？我看界面很好看，也想照着弄一下。

**Comment by Reverier-Xu** [comment]
是我自己做的主题 https://blog.woooo.tech/posts/ida_skins/ ，就在下一篇博客。

**Reply by mark0smith** [reply]
感谢分享，很好看👍
